### ISSUE TITLE
The main rules are confusing

### ISSUE THREAD

**Issue by Paul-Yuchao-Dong** [kurosawa4434/checkio-mission-unfair-districts#2]
This is a very interesting task and I found myself learning both politics and programming in this mission.

>There are two main rules of building districts. All districts should have same amount of people. All units should be connected inside of one district.

Yet the example given used `[[[2, 1], [1, 1], [1, 2]],[[2, 1], [1, 1], [0, 2]]]`, clearly not all districts have the same amount of people. In total there are 15 ppl, so how do I split 15 ppl among 6 districts so that each have 5 ppl? Clearly I am missing something important here.

>  All units should be connected inside of one district.

How is that expressed in the list?

> The same letters of string represents the same district.

But the first answer given was `['AAC','BBC]` what do A, B and C represent?

Looking forward to see the mission being more understandable. This is a great idea!

**Comment by oduvan** [comment]
> But the first answer given was ['AAC','BBC] what do A, B and C represent?

size of result list is the same as initial one. 

A, B and C are districts names.

if initial map is `[[[2, 1], [1, 1], [1, 2]],[[2, 1], [1, 1], [0, 2]]]` and result is `['AAC','BBC]` then we have 3 united districts, each of them has 2 units inside with following coordinates

A = 0,0 and 0,1
B = 1,0 and 1,1
C = 0,2 and 1,2

Does it make sense? Feel free to suggest a better explanation here. I would love to improve the mission.

**Comment by Paul-Yuchao-Dong** [comment]
Yes I think the explanation above is clear! I would suggest just put the above explanation into the mission description and people would get it! Now I think I can go ahead and solve it while having fun!

Thanks!

**Comment by Nataliia-Kholodkova** [comment]
Hi! Help please )) I don't understand where from do we take 'A', 'B' and 'C' ( if there are names of districts - how can we know, what units are included in district'A' or 'B'? Or i'm thinking in wron way?

**Comment by kurosawa4434** [comment]
> I don't understand where from do we take 'A', 'B' and 'C'

Please, you decide.　(e.g:  'a', 'b', 'c'... or '1', '2', '3'... ) .
The checker just regards the same letter as one district.

**Comment by Nataliia-Kholodkova** [comment]
Oh, can i describe my logic? 
I understand, that each grid in list [[[2, 1], [1, 1], [1, 2]],[[2, 1], [1, 1], [0, 2]]] (for exmple [2,1] or [1,1]) - is unit of one district? And 2,1 (sum 3) - amount people in unit? And we have to unite this units in district in way that in one district there were 5 peaple? (as in first example).  Am i right or i think in wrong way? If  i'm right, so 'A' is [2,1] (from first list in big list) and [2,1] (from second), 'C' is [1,1] and [0,1]? Or we shouldn't look at list(that in one big list there are few little lisls) and letters 'A' is [2, 1], [1, 1], 'B' is [1, 2],[2, 1] and '
C' is  [1, 1], [0, 2]? I can't understand where from we take this letters (or digits as you said). 
May be, i have problems with logic )))))))

**Comment by kurosawa4434** [comment]
How about these images ?
![unfair_districts_examples](https://user-images.githubusercontent.com/22162406/27608848-e9b757f4-5bc3-11e7-98c1-1988407ae75e.png)

The checker translate the answer: 
```
['AAC', 
 'BBC'] 
```
as
```
{'A': [grid[0][0], grid[0][1]],
 'B': [grid[1][0], grid[1][1]],
 'C': [grid[0][2], grid[1][2]]}
```

**Comment by Nataliia-Kholodkova** [comment]
I guess now i understand ) Thank you very much - visualization helps )

**Comment by oduvan** [comment]
@DahliaSR , maybe you can help us with improving mission description?

**Comment by oduvan** [comment]
Masayuki, thanks for help with visualisation.
Can you add it to the mission?

**Comment by Nataliia-Kholodkova** [comment]
May be, should write about this letters - as for me not clear where from they are. It confuses. 
"Input: Two arguments. Amount of people for each district (as integer) , grid of all electoral zone (as list of list (each unit has [A-peoples, B-peoples]))" - i thought, that 'AAC' means that grids are united in way A-candidate wins (two such grids) and B win. And 'C' is a districts with draw. This was wrong.
{'A': [grid[0][0], grid[0][1]],
 'B': [grid[1][0], grid[1][1]],
 'C': [grid[0][2], grid[1][2]]}
not very clear why 'A' and 'B' have some values in grids.
I don't know, how to describe this clear. And may be it's not clear for me, but other people understand.

**Comment by Nataliia-Kholodkova** [comment]
or maybe to give number every grid -  [grid[0][0]:'1', grid[0][1]-'2'.... and the answer for example ('1','4','5') and ('2','3','6')? No?

**Comment by kurosawa4434** [comment]
The first answer ['AAC','BBC]  represent:

 A | A | C
---|---|---
 B | B | C

The above has 3 districts and each district has a total of 5 people.
The three letters need only be different.

**Comment by Nataliia-Kholodkova** [comment]
I understand now. But i said, what in misiion's review confused me. 
Thank for explanation ) 

> The first answer ['AAC','BBC] represent:

A	A	C
B	B	C
The above has 3 districts and each district has a total of 5 people.
The three letters need only be different.

This is very helpful for understand )

**Comment by Paul-Yuchao-Dong** [comment]
I think some of the confusion may come from the fact that "A", "B" represented candidates, and ["A", "B", "C"], represented districts. 

if instead, the mission description used ["Republicans", "Democrats"] or ["R", D"] to represent the political candidates, and the mission is to gerrymander until Republican candidate wins. 

Also a bit more explanation for the expected output would help. I'm just throwing words out.

"The output is essentially labelling each district with a letter, districts with the same letter would become a post-gerrymander district. the population of the original districts with the same label would be merged into the new post-gerrymander district. The goal of the gerrymandering is to make sure that if the election were to be held on these new districts, Republican candidate would win."

"The constraint of gerrymandering is that you cannot label original districts that are not connected with each other with the same label."
